driver-sync/src/test/functional/com/mongodb/client/unified/UnifiedCrudHelper.java

```diff
@@ -1452,18 +1454,32 @@ OperationResult executeCreateSearchIndex(final BsonDocument operation) {
         BsonDocument arguments = operation.getDocument("arguments", new BsonDocument());
         BsonDocument model = arguments.getDocument("model");
         BsonDocument definition = model.getDocument("definition");
+        SearchIndexType type = getSearchIndexType(model.getString("type"));
 
         return resultOf(() -> {
             if (model.containsKey("name")) {
                 String name = model.getString("name").getValue();
-                collection.createSearchIndex(name, definition);
+                SearchIndexModel searchIndexModel = new SearchIndexModel(name, definition, type);
+                collection.createSearchIndexes(Collections.singletonList(searchIndexModel));
             } else {
-                collection.createSearchIndex(definition);
+                SearchIndexModel searchIndexModel = new SearchIndexModel(definition, type);
+                collection.createSearchIndexes(Collections.singletonList(searchIndexModel));
             }
             return null;
         });
     }
```

Comment:
Is there a reason why this method can't use `toIndexSearchModel`, and instead has to repeat what `toIndexSearchModel` does?